### PR TITLE
Generate documentation for GIT API

### PR DESCRIPTION
**Pull request by george-bit**
## Context
As part of intial work to build the GIT Bat Data Sharing API we need to generate docs for review by GiT

## Changes proposed in this pull request

This PR adds a lightweight mvc for the API Docs and defines a
 route at which these can be viewed. The structure of mirros that of the
data api docs.

Screenshot 2021-06-09 at 16 35 55

## Guidance to review

Check all is okay in the review app.

## Link to Trello card

https://trello.com/c/X8VGziRz/3491-generate-documentation-for-git-api

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/controllers/api_docs/candidates_api_docs/open_api_controller.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module APIDocs
+  module CandidatesAPIDocs
```
Comment (davidgisbey) [inline app/controllers/api_docs/candidates_api_docs/open_api_controller.rb:2]: Super nitty. I don't know why but `CandidateAPIDocs` feels right here. Interested to hear Duncans thoughts though i could def be wrong

File: config/candidates-api.yml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            "$ref": "#/components/schemas/CandidateRecord"
+    CandidateRecord:
+      type: object
+      required:
+        - id
+        - type
+        - data
+      properties:
+        id:
+          type: integer
```
Comment (davidgisbey) [inline config/candidates-api.yml:76]: this should be string

Comment (duncanjbrown) [reply config/candidates-api.yml:76]: Fixed in #4912 — this just needs a rebase

File: app/views/api_docs/candidates_api_docs/reference/reference.html.erb [hunk 3]
```diff
@@ -0,0 +1,59 @@
+<h1 class="govuk-heading-xl"><%= t('page_titles.api_docs.data_api.reference') %></h1>
```
Comment (duncanjbrown) [inline app/views/api_docs/candidates_api_docs/reference/reference.html.erb:1]: We need a new key here — currently reads "Apply for teacher training data API"

**Review by duncanjbrown — COMMENTED** [review]
Looking good! The JSON snippets are appearing unstyled — I think we need to add a clause to `ApplicationHelper#current_namespace` to make the header CSS includes work?

File: app/views/api_docs/candidate_api_docs/reference/reference.html.erb [hunk 4]
```diff
@@ -0,0 +1,59 @@
+<h1 class="govuk-heading-xl"><%= t('page_titles.api_docs.candidate_api.reference') %></h1>
+
+<p class="govuk-body">This is API documentation for the Apply for teacher training candidates API. This API provides candidate information to other services.</p>
```
Comment (duncanjbrown) [inline app/views/api_docs/candidate_api_docs/reference/reference.html.erb:3]: Let's nail this down now: is this the Candidate API or the Candidate**s** API?

Comment (george-bit) [reply app/views/api_docs/candidate_api_docs/reference/reference.html.erb]: Sorry @duncanjbrown I should have popped a note on the PR when I made the change. JAmes and I, decided to keep things uniform that we should make it the `Candidate API`. ☝️ this is just one I missed, all sorted now.

Comment (duncanjbrown) [reply app/views/api_docs/candidate_api_docs/reference/reference.html.erb]: Great 🎉

File: config/candidate-api.yml [hunk 5]
```diff
@@ -2,19 +2,19 @@
...
 servers:
 - description: Sandbox (test environment)
-  url: https://sandbox.apply-for-teacher-training.service.gov.uk/candidates-api
+  url: https://sandbox.apply-for-teacher-training.service.gov.uk/candidate-api
 - description: Production
-  url: https://www.apply-for-teacher-training.service.gov.uk/candidates-api
+  url: https://www.apply-for-teacher-training.service.gov.uk/candidate-api
 paths:
-  "/candidates":
+  "/candidate":
```
Comment (duncanjbrown) [inline config/candidate-api.yml:17]: I think it makes sense for this to be `/candidates` as is REST convention?

**Comment by duncanjbrown** [comment]
Looks like we have an issue with dates not rendering as [ISO8601](https://en.wikipedia.org/wiki/ISO_8601) in the docs, but this affects the other docs too — will add as a bug.

Screenshot 2021-06-14 at 10 05 04

**Comment by duncanjbrown** [comment]
Bug for dates: https://trello.com/c/VIDIqZFv/3510-api-docs-dont-render-timestamps-correctly

**Review by duncanjbrown — APPROVED** [review]
🚀

**Pull request merged**